### PR TITLE
add install target, which is expected by PyTorch.

### PR DESCRIPTION
**Pull request by erip**
## Motivation

Apparently pytorch_jni expects dependent libraries to have an install target. 

## Summary

Added install target with some Windows-specific install dirs. Based on [FBGEMM](https://github.com/pytorch/FBGEMM/blob/master/CMakeLists.txt#L240), which works in build.

### CODE REVIEW

**Comment by peterjc123** [comment]
@erip The variable `CMAKE_INSTALL_LIBDIR` relies on an external package. Please refer to https://github.com/pytorch/FBGEMM/blob/master/CMakeLists.txt#L6.

**Comment by peterjc123** [comment]
You'll then need to fix tje calls of `target_include_directories`. For more details,you may read this post. https://github.com/arvidn/libtorrent/issues/3101#issuecomment-396195787

**Comment by erip** [comment]
Thanks, @peterjc123  - I was fixing that up as you commented. 😄 looks like it worked ... enough. I'll update PyTorch to use my fork to test.

File: CMakeLists.txt [hunk 1]
```diff
@@ -39,8 +41,8 @@ set_target_properties(fbjni PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
 target_compile_options(fbjni PRIVATE ${FBJNI_COMPILE_OPTIONS})
 target_compile_options(fbjni PRIVATE -DBUILDING_FBJNI)
 
-target_include_directories(fbjni PUBLIC
-  cxx
+target_include_directories(fbjni BEFORE
+  PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/cxx> 
 )
 
 if (NOT ANDROID_ABI)
```
Comment (peterjc123) [inline CMakeLists.txt:48]: Should we also fix the usage of `target_include_directories` that is related to `JAVA_HOME` in the following lines? Actually, I don't think they should be in the installed configuration file.

Comment (erip) [reply CMakeLists.txt:48]: I'm honestly not sure. ☹️ This is all a bit magic to me, so that may be a question for @dreiss.

**Comment by erip** [comment]
Tracking the PyTorch impact [here](https://github.com/pytorch/pytorch/pull/46466). 😄

File: CMakeLists.txt [hunk 2]
```diff
@@ -100,3 +102,14 @@ if (ANDROID_ABI)
...
+install(TARGETS fbjni EXPORT fbjniLibraryConfig
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}) #For windows
+
+install(EXPORT fbjniLibraryConfig DESTINATION share/cmake/fbjni
+  FILE fbjniLibraryConfig.cmake)
+
+if(MSVC)
+  install(TARGETS fbjni DESTINATION ${CMAKE_INSTALL_LIBDIR})
```
Comment (peterjc123) [inline CMakeLists.txt:114]: One thing you are missing is that you forget to install the PDB fles on debug builds. See https://github.com/pytorch/FBGEMM/blob/master/CMakeLists.txt#L253-L255

**Comment by erip** [comment]
FWIW, it all looks good on the PyTorch end.

**Comment by erip** [comment]
also @mhorowitz may be a good person to review this

**Comment by erip** [comment]
Any thoughts here, @dreiss?

**Comment by erip** [comment]
Can you please review, @dreiss or @passy? This is an incredibly small change - it should take no more than 60s.

**Comment by erip** [comment]
@swolchok would you mind reviewing this PR?

**Comment by swolchok** [comment]
Sorry, I'm not at all familiar with fbjni's build system.

**Comment by dreiss** [comment]
Hey, I'm pretty focused now on getting some stuff ready for PyTorch Dev Day, but I should be able to take a look at this next week.

I don't agree that this will take 60s to review.  It's calling a lot of external functions, and I need to understand them in order to think through possible unintended side effects.  Each one will take more than 60 seconds.

**Comment by erip** [comment]
Apologies for the underestimation of required effort. I was under the impression that adding `install()` targets strictly impacted the case where `--config target` is used (which was previously undefined and thus in a default empty configuration state). 

Adding relevant documentation here in case it's helpful:

- https://cmake.org/cmake/help/latest/command/install.html
- https://cmake.org/cmake/help/v3.0/module/GNUInstallDirs.html

**Comment by erip** [comment]
I can't see what the linter issue is, but I'm happy to fix it if someone can point me in the right direction.

**Comment by erip** [comment]
Is this waiting on anything, @dreiss?

**Pull request closed without merging**